x-pack/examples/ui_actions_enhanced_examples/public/dashboard_to_discover_drilldown/components/discover_drilldown_config/discover_drilldown_config.tsx

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import { EuiFormRow, EuiSelect, EuiSwitch, EuiSpacer } from '@elastic/eui';
+import { txtChooseDestinationIndexPattern } from './i18n';
+
+export interface IndexPatternItem {
+  id: string;
+  title: string;
+}
+
+export interface DiscoverDrilldownConfigProps {
+  activeIndexPatternId?: string;
+  indexPatterns: IndexPatternItem[];
+  onIndexPatternSelect: (indexPatternId: string) => void;
+  customIndexPattern?: boolean;
+  onCustomIndexPatternToggle?: () => void;
+  carryFiltersAndQuery?: boolean;
+  onCarryFiltersAndQueryToggle?: () => void;
+  carryTimeRange?: boolean;
+  onCarryTimeRangeToggle?: () => void;
+}
+
+export const DiscoverDrilldownConfig: React.FC<DiscoverDrilldownConfigProps> = ({
+  activeIndexPatternId,
+  indexPatterns,
+  onIndexPatternSelect,
+  customIndexPattern,
+  onCustomIndexPatternToggle,
+  carryFiltersAndQuery,
+  onCarryFiltersAndQueryToggle,
+  carryTimeRange,
+  onCarryTimeRangeToggle,
+}) => {
+  return (
+    <>
+      {!!onCustomIndexPatternToggle && (
+        <>
+          <EuiFormRow hasChildLabel={false}>
+            <EuiSwitch
+              name="customIndexPattern"
+              label="Use custom index pattern"
+              checked={!!customIndexPattern}
+              onChange={onCustomIndexPatternToggle}
+            />
+          </EuiFormRow>
+          {!!customIndexPattern && (
+            <EuiFormRow fullWidth label={txtChooseDestinationIndexPattern}>
+              <EuiSelect
+                name="selectDashboard"
+                hasNoInitialSelection={true}
+                fullWidth
+                options={[
+                  { id: '', text: 'Pick one...' },
+                  ...indexPatterns.map(({ id, title }) => ({ value: id, text: title })),
+                ]}
+                value={activeIndexPatternId || ''}
+                onChange={e => onIndexPatternSelect(e.target.value)}
+              />
+            </EuiFormRow>
+          )}
+          <EuiSpacer size="xl" />
+        </>
+      )}
+
+      {!!onCarryFiltersAndQueryToggle && (
+        <EuiFormRow hasChildLabel={false}>
+          <EuiSwitch
```

Comment:
No, it does not, this is a sample drilldown in the example plugin. There is quite some work needed to make it work, not something I wanted to do in an example. And when I was making this example drilldown, we did not have [filter creation utils](https://github.com/elastic/kibana/pull/62969), nor we still have URL generator for Discover app (I think).

But I think it is useful to have this UI as developers can see in the flyout how a drilldown could look like; and when we will be actually building a proper Discover drilldown this could will be copy-pasted over and connected to filtering.

FWIW, there is Discover drilldown task: https://github.com/elastic/kibana/issues/60227